### PR TITLE
Update vulnerability detector tests to 3.13

### PR DESCRIPTION
**Pull request by jmv74211**
Hello team,

Due to new changes in log messages and in the internal functioning of vulnerability detector module, it has been necessary to adapt the callbacks regex and make some additional changes.

This PR adds the necessary changes to update the vulnerability detector tests from `3.12.3` to `3.13`.

The current result of 3.13 branch would be:

```
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1, testinfra-5.0.0
collected 83 items                                                                                     

test_vulnerability_detector/test_general_settings/test_general_settings_enabled.py .ss.          [  4%]
test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py ...       [  8%]
test_vulnerability_detector/test_general_settings/test_general_settings_interval.py ............ [ 22%]
                                                                                                 [ 22%]
test_vulnerability_detector/test_general_settings/test_general_settings_run_on_start.py ..       [ 25%]
test_vulnerability_detector/test_providers/test_download_feeds.py .........                      [ 36%]
test_vulnerability_detector/test_providers/test_providers_enabled.py ........                    [ 45%]
test_vulnerability_detector/test_providers/test_providers_os.py .........                        [ 56%]
test_vulnerability_detector/test_providers/test_providers_update_from_year.py ................   [ 75%]
test_vulnerability_detector/test_providers/test_update_interval.py ................              [ 95%]
test_vulnerability_detector/test_scan_results/test_redhat_inventory_redhat_feeds.py ....         [100%]

============================== 81 passed, 2 skipped in 1820.88s (0:30:20) ==============================
```

Best regards.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py [hunk 1]
```diff
@@ -44,7 +56,8 @@ def callback_detect_vulnerability_scan_sleeping(line):
 
 
 def callback_detect_update_from_year_result(line):
-    invalid_year_msg = r'.*wazuh-modulesd.*wmodules-vuln-detector.*ERROR: Invalid content for \'update_from_year\' option at module \'vulnerability-detector\''
+    invalid_year_msg = r'.*wazuh-modulesd.*wmodules-vuln-detector.*ERROR: Invalid content for \'update_from_year\'\
+                         option at module \'vulnerability-detector\''
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/vulnerability_detector.py:60]:
```suggestion
    invalid_year_msg = r".*wazuh-modulesd.*wmodules-vuln-detector.*ERROR: Invalid content for 'update_from_year'\
                         option at module 'vulnerability-detector'"
```

File: tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py [hunk 2]
```diff
@@ -28,72 +27,68 @@
...
+                            callback=custom_callback_vulnerability,
+                            error_message='Alert did not appear at the start of the test')
+
+    # Check if alert does not appear during ignore time
+    for i in range(1, jumps):
+        check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
+        with pytest.raises(TimeoutError):
+            wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
+                                    callback=custom_callback_vulnerability)
+            raise AttributeError('Alert appeared before ignore_time was finished')
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py:86]: In loops like this where you don't use the variable `i` you can just do:
```suggestion
    for _ in range(1, jumps):
        check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
        with pytest.raises(TimeoutError):
            wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
                                    callback=custom_callback_vulnerability)
            raise AttributeError('Alert appeared before ignore_time was finished')
```

File: tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py [hunk 3]
```diff
@@ -88,19 +86,23 @@ def test_download_feeds(get_configuration, configure_environment, restart_module
     """
     check_apply_test({'test_download_feeds'}, get_configuration['tags'])
 
-    provider = get_configuration['metadata']['provider_name']
     os = (get_configuration['metadata']['os']).capitalize()
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py:89]:
```suggestion
    os = get_configuration['metadata']['os'].capitalize()
```

File: tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py [hunk 4]
```diff
@@ -88,19 +86,23 @@ def test_download_feeds(get_configuration, configure_environment, restart_module
...
     system_log = get_configuration['metadata']['system_log']
     download_timeout = get_configuration['metadata']['download_timeout']
+    callback_system_log = system_log
+
+    # If provider configuration has <os> tag
+    if os:
+        callback_system_log = f"{system_log} {os}"
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py:96]: This code may trigger a KeyError exception. If there isn't any `os` tag in the configuration, line 89 will raise the exception.

```suggestion
    system_log = get_configuration['metadata']['system_log']
    download_timeout = get_configuration['metadata']['download_timeout']

    # If provider configuration has <os> tag
    try:
        callback_system_log = f"{system_log} {get_configuration['metadata']['os'].capitalize()}"
    except KeyError:
        callback_system_log = system_log
```

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_enabled.py [hunk 5]
```diff
@@ -42,35 +45,30 @@
...
 
 # fixtures
-
 @pytest.fixture(scope="module", params=configurations, ids=ids)
 def get_configuration(request):
     """Get configurations from the module."""
     return request.param
 
 
+# tests
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_providers_enabled.py:58]: This comment is unnecessary.

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py [hunk 6]
```diff
@@ -43,35 +47,30 @@
...
 
 # fixtures
-
 @pytest.fixture(scope="module", params=configurations, ids=ids)
 def get_configuration(request):
     """Get configurations from the module."""
     return request.param
 
 
+# tests
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py:60]: This one neither.

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_update_from_year.py [hunk 7]
```diff
@@ -42,24 +45,24 @@
...
+
 # fixtures
 @pytest.fixture(scope='module', params=configurations, ids=ids)
 def get_configuration(request):
     """Get configurations from the module."""
     return request.param
 
 
-# Tests
+# tests
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_providers_update_from_year.py:56]: And this one.

File: tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py [hunk 8]
```diff
@@ -52,12 +51,14 @@
...
 
+
 # Fixtures
 @pytest.fixture(scope='module', params=configurations, ids=ids)
 def get_configuration(request):
     """Get configurations from the module."""
     return request.param
 
+
 # Tests
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py:62]: This one too.

File: tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py [hunk 9]
```diff
@@ -71,15 +72,15 @@ def test_update_interval(get_configuration, configure_environment, restart_modul
...
-        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+        wazuh_log_monitor.start(timeout=VULN_DETECTOR_GLOBAL_TIMEOUT,
                                 callback=callback_provider_database_updating)
         raise AttributeError(f'Unexpected event {provider} database updating')
 
     # Travels the time set in the update interval parameter
     check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
 
-    # Check that the feed is downloaded after the set update interval
+    # Check that the feed is downloaded after the set update interval(10s for vuln thread refresh + 5s to wait callback)
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py:82]: This comment can be split into two different lines.

File: tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py (hunk 4 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py:96]: It is not necessary, there will always be an `os`

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py (hunk 6 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py]: Done in 7fcdc7ebecdd546a4957814ec960f6304685d265

File: tests/integration/test_vulnerability_detector/test_providers/test_providers_update_from_year.py (hunk 7 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_providers_update_from_year.py]: Done in 7fcdc7ebecdd546a4957814ec960f6304685d265

File: tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py (hunk 8 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py]: Done in 7fcdc7ebecdd546a4957814ec960f6304685d265

File: tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py (hunk 9 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_update_interval.py]: Done in 77223ed24d8fff611947e817d180bf744720ed4c

File: tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py (hunk 3 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_providers/test_download_feeds.py]: Done in 77223ed24d8fff611947e817d180bf744720ed4c

File: tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py (hunk 2 above)
Comment (jmv74211) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py]: Done in 77223ed24d8fff611947e817d180bf744720ed4c

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py (hunk 1 above)
Comment (jmv74211) [reply deps/wazuh_testing/wazuh_testing/vulnerability_detector.py]: Done in 77223ed24d8fff611947e817d180bf744720ed4c

**Pull request merged**